skbio/io/format/genbank.py

```diff
@@ -0,0 +1,865 @@
+from skbio.util._misc import chunk_str
+from skbio.sequence import Sequence, DNA, RNA, Protein
+
+
+genbank = create_format('genbank')
+
+# date format in locus line of genbank record
+_TIME_FORMAT = '%d-%b-%Y'
+# This list is ordered
+# used to read and write genbank file.
+_HEADERS = ['LOCUS',
+            'DEFINITION',
+            'ACCESSION',
+            'VERSION',
+            'DBSOURCE',
+            'DBLINK',
+            'KEYWORDS',
+            'SOURCE',
+            'REFERENCE',
+            'COMMENT',
+            'FEATURES',
+            'ORIGIN']
+
+
+@genbank.sniffer()
+def _genbank_sniffer(fh):
+    # check the 1st real line is a valid LOCUS line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        _parse_locus([line])
+    except GenBankFormatError:
+        return False, {}
+    return True, {}
+
+
+@genbank.reader(None)
+def _genbank_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_genbanks(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@genbank.reader(Sequence)
+def _genbank_to_sequence(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, Sequence, **kwargs)
+
+
+@genbank.reader(DNA)
+def _genbank_to_dna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, DNA, **kwargs)
+
+
+@genbank.reader(RNA)
+def _genbank_to_rna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, RNA, **kwargs)
+
+
+@genbank.reader(Protein)
+def _genbank_to_protein(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, Protein, **kwargs)
+
+
+@genbank.writer(None)
+def _generator_to_genbank(obj, fh):
+    for obj_i in obj:
+        _serialize_single_genbank(obj_i, fh)
+
+
+@genbank.writer(Sequence)
+def _sequence_to_genbank(obj, fh):
+    _serialize_single_genbank(obj, fh)
+
+
+@genbank.writer(DNA)
+def _dna_to_genbank(obj, fh):
+    _serialize_single_genbank(obj, fh)
+
+
+@genbank.writer(RNA)
+def _rna_to_genbank(obj, fh):
+    _serialize_single_genbank(obj, fh)
+
+
+@genbank.writer(Protein)
+def _protein_to_genbank(obj, fh):
+    _serialize_single_genbank(obj, fh)
+
+
+def _construct(record, constructor=None, **kwargs):
+    '''Construct the object of Sequence, DNA, RNA, or Protein.
+    '''
+    seq, md, pmd = record
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            # RNA mol type has T instead of U for genbank from from NCBI
+            constructor = DNA
+        elif unit == 'aa':
+            constructor = Protein
+        else:
+            constructor = Sequence
```

Comment:
I think there needs to be an `elif constructor is RNA:` branch where we will construct a `DNA`, but then return the transcription of it.